### ISSUE TITLE
Add back iOSVersionStyle option to CupertinoActivityIndicator to allow catering to iOS13 and iOS14

### ISSUE THREAD

**Issue by acoutts** [flutter/flutter#68285]
## Use case
In Flutter 1.23, the `iOSVersionStyle` was removed from `CupertinoActivityIndicator` and it now only displays the iOS 14 style indicator. This means all Flutter iOS apps will use the iOS 14 style activity indicator even if the device is not using iOS 14, which doesn't match the native experience.

Instead we should be able to specify the indicator style based on the device to match the native experience. Since Flutter still has support for <iOS 14 (currently iOS 9.0 is the minimum), we should support the native experience on those versions of iOS too, and not make assumptions about the device OS version based on what Flutter version compiled the app.

## Proposal
Allow manually overriding `iOSVersionStyle` for the iOS 13 or iOS 14 activity indicator style. Developers can use the `device_info` package to detect the iOS version of the device and display the correct indicator style to the user.

This is how it worked prior to Flutter 1.23:
```dart
CupertinoActivityIndicator(
  iOSVersionStyle: CupertinoActivityIndicatorIOSVersionStyle.iOS13,
)
```

**Comment by acoutts** [comment]
cc https://github.com/flutter/flutter/issues/61047 for general iOS 14 tracking.

**Comment by lukepighetti** [comment]
I still don't understand why we have to specify this in constructor arguments. Shouldn't CupertinoActivityIndicator change based on the iOS version on its own?

**Comment by acoutts** [comment]
> I still don't understand why we have to specify this in constructor arguments. Shouldn't CupertinoActivityIndicator change based on the iOS version on its own?

That would be the most ideal behavior I think too.

**Comment by DanielEdrisian** [comment]
I believe this should be the standard approach. Will tackle this soon. Thanks @acoutts @lukepighetti

**Comment by DanielEdrisian** [comment]
@ctrysbita Would you be able to re-add the iOS version that was deprecated in https://github.com/flutter/flutter/pull/62741?

**Comment by ctrysbita** [comment]
@DanielEdrisian Hi, according to the disscussion in #62521 , we've decided to remove the old style.

**Comment by lukepighetti** [comment]
You're just throwing out work every time there is an iOS release. These things can be built in such a way that bit rot will be minimal.

**Comment by ctrysbita** [comment]
\cc @justinmc @jmagman

**Comment by justinmc** [comment]
My main reason for opposing Flutter's support of multiple OS version styles has essentially been a lack of developer resource, which I posted about here: https://github.com/flutter/flutter/issues/62521#issuecomment-666461667

However this is probably up to @xster and/or anyone else that would put in the work to build and maintain this.

**Comment by lukepighetti** [comment]
I think there's a lot to unpack in your comment.

1) Flutter does have to support the latest iOS version style, afaik.
2) Flutter does not have to actively develop legacy iOS widget.
3) Deleting old efforts makes them wasted efforts.

So whenever Apple updates an iOS design that we have represented as a widget, why not build the new one, keep the old one, and let it automatically change based on the host iOS version?

**Comment by acoutts** [comment]
I agree. It's not like Apple is going to ever change the iOS 13 style indicator now, so the upkeep should be minimal to none. It's really not acceptable to assume everyone on Flutter 1.23 is using iOS 14.

**Comment by SolitaryWolf** [comment]
I need to add back iosVersionStyle option too.

**Comment by Piinks** [comment]
I am going to close this since we will not be adding back these older design elements, the style guide discusses our goal to match the latest, rather than maintaining older variants. https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#widget-libraries-follow-the-latest-oem-behavior
